### DISCUSSION TITLE
datepicker multiple clicks needed

### DISCUSSION THREAD

**Discussion by allard-innocom** [BlazorFluentUI/BlazorFluentUI#329] [Q&A] [answered]
I am using the date picker and in my application multiple clicks (sometimes 3 !) are needed for the date picker popup to appear.

Something similar happens on the demo page: https://www.blazorfluentui.net/datePickerPage
If I click on "select a date" in the basic datepicker, nothing happens or sometimes only the field gets highlighted with a blue border.  The behaviour is not consistent. (refresh the page, and try it out)

It does not seem to be a timing issue. It also happens after waiting a few seconds.

When I first click any other field, and THEN click on "select a date", it appears immediately.

Is this a bug ?

I would like the datepicker to appear on the first click on the field.

**Comment by vnbaaij** [comment, marked as answer]
This is fixed in V5.3.7

**Reply by allardwp** [reply]
unfortunately, I can still reproduce the same issue https://www.blazorfluentui.net/datePickerPage (is says it is version 5.3.5.0).

To reproduce: 
* click on the "select a date.."  placeholder. 
* field gets a blue border but popup does not open
* click another time, now it opens

Clicking on the "icon"  works sometimes, but there are other times that nothing happens. The field does not even get a blue border.  At other times the datepicker opens immediately.

Try it out by refreshing the page (F5) each time

**Reply by vnbaaij** [reply]
Sorry, I was mistaking this with the Calendar page where I did fix something. Looking at it again now...

**Reply by limefrogyank** [reply]
I think it was a case of the javascript module loading taking too long in CalloutContent's OnAfterRenderAsync method.  A 2nd call is made to this method, but it's no longer `firstRender`, yet `isEventHandlersRegistered` is still false... so it flips that flag, then `CalculateCalloutPositionAsync` method is never called because it's not firstRender.

**Reply by vnbaaij** [reply]
Good catch! Works as it should now.
